SimG4Core/Generators/src/Generator.cc

```diff
@@ -495,21 +495,14 @@ bool Generator::particlePassesPrimaryCuts(const G4ThreeVector &p) const {
   return flag;
 }
 
-bool Generator::isExotic(int pdgcode) const {
-  int pdgid = std::abs(pdgcode);
-  return ((pdgid >= 1000000 && pdgid < 4000000) ||  // SUSY, R-hadron, and technicolor particles
-          pdgid == 17 ||                            // 4th generation lepton
-          pdgid == 34 ||                            // W-prime
-          pdgid == 37)                              // charged Higgs
-             ? true
-             : false;
-}
-
-bool Generator::isExoticNonDetectable(int pdgcode) const {
+bool Generator::isChargedExotic(int pdgcode) const {
   int pdgid = std::abs(pdgcode);
   HepPDT::ParticleID pid(pdgcode);
   int charge = pid.threeCharge();
-  return ((charge == 0) && (pdgid >= 1000000 && pdgid < 1000040))  // SUSY
```

Comment:
@weishi10141993 , here we need to have some logic allowing neutral be selected.

Reply:
Some other group (PR #20083) specifically do not want to select neutral SUSY particles. They use !isExoticNonDetectable to exclude SUSY particles from what PR #16366 specified.

Do you mean we should allow to select neutral particles from 1000040 to 4000000; and for 1000000 to 1000040, only select charged?